strawberry_django/mutations/resolvers.py

```diff
@@ -216,19 +216,116 @@ def create(
     full_clean: bool | FullCleanOptions = True,
     pre_save_hook: Callable[[_M], None] | None = None,
 ):
+    # Before creating your instance, verify this is not a bulk create
+    # if so, add them one by one. Othewise, get to work.
     if isinstance(data, list):
         return [create(info, model, d, full_clean=full_clean) for d in data]
 
+    # Also, the appraoch below will use the manager to create the instnace
+    # rather then manually creating it.  If you have a pre_save_hook
+    # use the update method instead.
+    if pre_save_hook:
+        return update(
+            info,
+            model(),
+            data,
+            full_clean=full_clean,
+            pre_save_hook=pre_save_hook,
+        )
+
+    # We will use a dummy-instance to trigger form validation
+    # However, this instance should not be saved as it will
+    # circumvent the manager create method.
+    dummy_instance = model()
+
+    fields = get_model_fields(model)
+    files: list[
+        tuple[
+            models.FileField,
+            Upload | Literal[False],
+        ]
+    ] = []
+    m2m: list[tuple[ManyToManyField | ForeignObjectRel, Any]] = []
+    create_kwargs = {}
+
     if dataclasses.is_dataclass(data):
-        data = vars(cast(object, data))
+        data = vars(data)
+
+    for name, value in data.items():
```

Comment:
This is mostly duplicating the code from `update` into create.

I do understand why, but can we maybe move some of the duplicated code to a separated utility function and call it in both places?

Reply:
I'll have a look to clean that up once we've managed to find the source of the last remaining issue.
Why does the model content-type get reverted to its parent when the object is not present in the qs.

Reply:
I've done a few things:
- cleaned up the repeated code
- restructured the update a little to ensure a single flow of data rather then trying to accommodate single and multiple instances
- moved dataclasses into a separate file

Ideally I wanted the new helper in another file, but there was just too much circular imports that made it quite pointless.